packages/mongodb-explain-compat/lib/index.js

```diff
@@ -72,18 +72,56 @@ function mapStages(queryPlan, sbeExecutionStages) {
   });
 }
 
-module.exports = function(explain) {
-  explain = JSON.parse(JSON.stringify(explain));
-  if ((explain.explainVersion || 0) < 2) {
+function isAggregationExplain(explain) {
+  return !!explain.stages;
+}
+
+/**
+ * https://wiki.corp.mongodb.com/display/QUERY/Explain+Notes
+ */
+function mapAggregationExplain(explain) {
+  // In aggregations for SBE, the cursor based stage is always first.
+  const firstStage = explain.stages[0]; // {$cursor: {queryPlanner: {}, executionStats: {} }}
+  const firstStageKey = Object.keys(firstStage)[0]; // $cursor
+  if (!firstStageKey.match(/cursor/)) {
```

Comment:
Just curious, is there a specific reason for the [0] and the regular expression here? I don’t think the server would typically guarantee ordering of keys, so if they added a key before `cursor`, this would not match anymore.

Reply:
awesome! fixed.

Reply:
I’m still not quite sure about the regular expression part – is there any other key than `$cursor` or maybe `cursor` without the `$` that we need to take care of? https://github.com/mongodb/mongo/blob/e9e4e99d0ff87ec9b6ed2a689a4c42cdba7f5527/src/mongo/shell/explainable.js#L607 has a literal check as well

Reply:
Yeah! The first stage is a general cursor stage and it be $cursor, $geoNearCursor (i know so far these two only, but i assume there are more). 

I had a chat with the team in #aggregation-framework regarding this and accordingly updated the documentation (linked in comments of code).

Reply:
Yeah, I see – feels like the regexp could be tightened up a bit to `/^\$.*cursor/i` if the distinguishing feature of the key is that it starts with a `$`, but this makes sense :+1: Might be good to leave a comment about this?